grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -546,7 +546,15 @@ class Product implements Comparable, Serializable {
 
 
     Date latestInventoryDate(String locationId) {
-        return latestTransactionDate(locationId, [TransactionCode.PRODUCT_INVENTORY])
+        // We want to look for transaction types, not only codes
+        // so that we don't take into account some types with CREDIT code
+        // that we wouldn't want to include
+        List<String> transactionTypeIds = [
+                Constants.ADJUSTMENT_CREDIT_TRANSACTION_TYPE_ID,
+                Constants.PRODUCT_INVENTORY_TRANSACTION_TYPE_ID,
+                Constants.INVENTORY_BASELINE_TRANSACTION_TYPE_ID
```

Comment:
I think this one is unnecessary:
```
Constants.INVENTORY_BASELINE_TRANSACTION_TYPE_ID
```
because it's already included in 
```
Constants.PRODUCT_INVENTORY_TRANSACTION_TYPE_ID,
```